src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/KeyworkerStatsService.java

```diff
@@ -368,33 +378,36 @@ private PrisonStatsDto getPrisonStatsDto(String prisonId, CalcDateRange range, P
 
         final TemporalAdjuster weekAdjuster = TemporalAdjusters.previousOrSame(DayOfWeek.SUNDAY);
 
-        SortedMap<LocalDate, Long> kwSummary = new TreeMap<>(dailyStats.stream().collect(
-                Collectors.groupingBy(s -> s.getSnapshotDate().with(weekAdjuster),
-                        Collectors.summingLong(PrisonKeyWorkerStatistic::getNumberKeyWorkerSessions))
-        ));
-
-        TreeMap<LocalDate, BigDecimal> compliance = new TreeMap<>(dailyStats.stream().collect(
-                Collectors.groupingBy(s -> s.getSnapshotDate().with(weekAdjuster),
-                        Collectors.averagingDouble(p ->
-                        {
-                            int projectedKeyworkerSessions = Math.floorDiv(p.getTotalNumPrisoners(), prisonConfig.getKwSessionFrequencyInWeeks() * 7);
-                            return getComplianceRate(p.getNumberKeyWorkerSessions(), projectedKeyworkerSessions).doubleValue();
-                        }))
-        ).entrySet().stream().filter(e -> e.getValue() != null).collect(Collectors.toMap(Map.Entry::getKey,
-                e -> new BigDecimal(e.getValue()).setScale(2, RoundingMode.HALF_UP))));
-
-        int avgOverallKeyworkerSessions = (int)Math.floor(kwSummary.values().stream().collect(averagingDouble(p -> p)));
-
-        return PrisonStatsDto.builder()
+        var prisonStats = PrisonStatsDto.builder()
                 .requestedFromDate(range.getStartDate())
                 .requestedToDate(range.getEndDate())
                 .current(current)
                 .previous(previous)
-                .keyworkerSessionsTimeline(kwSummary)
-                .avgOverallKeyworkerSessions(avgOverallKeyworkerSessions)
-                .complianceTimeline(compliance)
-                .avgOverallCompliance(getAverageCompliance(compliance.values()))
                 .build();
+
+        if (dailyStats != null) {
+            var kwSummary = new TreeMap<>(dailyStats.stream().collect(
+                    Collectors.groupingBy(s -> s.getSnapshotDate().with(weekAdjuster),
+                            Collectors.summingLong(PrisonKeyWorkerStatistic::getNumberKeyWorkerSessions))
+            ));
+            prisonStats.setKeyworkerSessionsTimeline(kwSummary);
```

Comment:
this looks odd - you are using the builder to create prisonStats and then calling the setters in the object.
Better to create var of the builder and then call the methods on that here

Reply:
@petergphillips - duh! - changed it now